oap-ws-openapi/src/main/java/oap/ws/openapi/OpenapiGenerator.java

```diff
@@ -163,21 +183,29 @@ public Result processWebservice( Class clazz, String context ) {
         return Result.PROCESSED_OK;
     }
 
-    private Operation prepareOperation( Reflection.Method method, WsMethodDescriptor wsDescriptor, OpenAPI api, Tag tag ) {
+    private Operation prepareOperation( Reflection.Method method,
+                                        WsMethodDescriptor wsMethodDescriptor,
+                                        WsSecurityDescriptor wsSecurityDescriptor,
+                                        Tag tag ) {
         var params = Lists.filter( method.parameters, WsApiReflectionUtils::filterParameter );
         var returnType = prepareType( method.returnType() );
 
         Operation operation = new Operation();
         operation.addTagsItem( tag.getName() );
-        operation.setOperationId( wsDescriptor.id );
+        operation.setOperationId( wsMethodDescriptor.id );
         operation.setParameters( prepareParameters( params ) );
-        operation.description( wsDescriptor.description );
-        operation.setRequestBody( prepareRequestBody( params, api ) );
-        operation.setResponses( prepareResponse( returnType, wsDescriptor.produces, api ) );
+        operation.description( wsMethodDescriptor.description );
+        operation.setRequestBody( prepareRequestBody( params ) );
+        operation.setResponses( prepareResponse( returnType, wsMethodDescriptor.produces ) );
+        Optional<WsDeprecated> deprecated = method.findAnnotation( WsDeprecated.class );
+        deprecated.ifPresent( x -> operation.deprecated( true ) );
+        if ( wsSecurityDescriptor != WsSecurityDescriptor.NO_SECURITY_SET ) {
+            operation.addSecurityItem( new SecurityRequirement().addList( "JWT" ) );
```

Comment:
need to publish permissions from WsSecurity

Reply:
To here:

Method test1
	[GET] /example/test/sort/{id}
	Produces application/json
	Realm param <unsecure>
	Permissions <unsecure>
	Returns String
	Parameters
		id: path String

?

Reply:
done